### PR TITLE
 environment variables

### PR DESCRIPTION
**Pull request by tboulogne**
I want to merge this change because...
 DATABASE_STRING AND TIME_ZONE  are scratch on git pull.

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] Database migration files are up to date.
1. [ ] The changes are tested.
1. [ ] GraphQL schema and type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for 71b1929f5c70978d8cc919284df602cccbc46eba (tboulogne/saleor @ master)
Base comparison is dbf12995d396097df457b2f40a3e66641f8c9993.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  add billing address to checkout            	         34	         34	             20
  add shipping to checkout                   	          7	          7	              0
  checkout payment charge                    	         10	         10	              0
  complete checkout                          	          8	          8	              0
  create checkout                            	         50	         50	             24

# api.benchmark homepage
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  retrieve main menu                         	          5	          5	              0
  retrieve product list                      	          4	          4	              0
  retrieve secondary menu                    	          5	          5	              0
  retrieve shop                              	          2	          2	              0

# api.benchmark product
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product details                            	         14	         14	              2
  retrieve product attributes                	          9	          9	              0

# api.benchmark variant
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product variant bulk create                	         51	         51	              3
  retrieve variant list                      	         16	         16	              6

# api product sorting attributes
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  sort product not having attribute data     	         21	         21	              0
```

</p>
</details>

File: saleor/settings.py [hunk 1]
```diff
@@ -48,12 +48,13 @@ def get_bool_from_env(name, default_value):
 
 DATABASES = {
     "default": dj_database_url.config(
-        default="postgres://saleor:saleor@localhost:5432/saleor", conn_max_age=600
+        default=os.environ.get("DATABASE_STRING", "postgres://saleor:saleor@localhost:5432/saleor"),
```
Comment (maarcingebala) [inline saleor/settings.py]: We don't need this, there is already `DATABASE_URL` env variable.

File: saleor/settings.py [hunk 2]
```diff
@@ -48,12 +48,13 @@ def get_bool_from_env(name, default_value):
...
     "default": dj_database_url.config(
-        default="postgres://saleor:saleor@localhost:5432/saleor", conn_max_age=600
+        default=os.environ.get("DATABASE_STRING", "postgres://saleor:saleor@localhost:5432/saleor"), 
+        conn_max_age=600
     )
 }
 
 
-TIME_ZONE = "America/Chicago"
+TIME_ZONE = os.environ.get("TIME_ZONE", "America/Chicago")
```
Comment (maarcingebala) [inline saleor/settings.py]: What is your use case for setting `TIME_ZONE` through env? You have different time zones in different instances/environments?

Comment (tboulogne) [reply saleor/settings.py]: @maarcingebala my TIME_ZONE is "Europe/Paris"
The goal is to left untouched settings.py between git pull.

Could you explain how to override this DATABASE_URL please ?

Comment (maarcingebala) [reply saleor/settings.py]: It's an environment variable, you would use it exactly like the `DATABASE_STRING` that you proposed. But I think the solution that you really need is local settings file that is not tracked in Git.

In your repo, you should create a `local_settings.py` next to `settings.py`. Inside you need to import all your standard settings:

```python
# local_settings.py

from .settings import *

# here you can override any Django settings e.g.:

TIME_ZONE = "Europe/Paris"
```

When you have it, run your project with `DJANGO_SETTINGS_MODULE` pointing to that settings file e.g. `DJANGO_SETTINGS_MODULE=saleor.local_settings`. And that's it - when you run the project it should use your local settings file which is not tracked by Git and you can do there whatever you need for local configuration.

**Comment by maarcingebala** [comment]
Closing this since the same can be achieved with `local_settings.py` and `DJANGO_SETTINGS_MODULE=saleor.local_settings`.

File: saleor/settings.py (hunk 2 above)
Comment (tboulogne) [reply saleor/settings.py]: @maarcingebala thanks for this tips :-)

**Pull request closed without merging**